### PR TITLE
Added support for the batch server API.

### PR DESCRIPTION
**Pull request by n1k0**
This only supports batch article creation for now, but the diff is large enough already, and adding other methods should be straightforward. 

Requesting feedback/review from @ametaireau @leplatrem @natim

### CODE REVIEW

File: src/js/batch.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   * Enqueue one or more article creation requests.
+   * @param  {...Object} entities Article entities.
+   * @return {Batch}
+   */
+  createArticle(...entities) {
+    this.requests = this.requests.concat(...entities.map(function(entity) {
+      // XXX could we factorize request definition with API methods? how?
+      return {
+        method: "POST",
+        path: "/v0/articles", // XXX repeating prefix for now to avoid 307
```
Comment (leplatrem) [inline src/js/batch.js]: https://github.com/mozilla-services/readinglist/issues/113

File: src/js/flux.js [hunk 2]
```diff
@@ -108,6 +109,26 @@ export var ArticleStore = DocBrown.createStore({
...
+    // We use the batch API mostly for testing purpose; we could simply append
+    // sample articles to the current list instead :)
+    var batch = this.api.createBatch();
+    return batch.createArticle.apply(batch, sampleArticles).process();
+  },
+
+  importSuccess: function(response) {
+    // XXX We want proper notification messages displayed to the end user here
+    response.responses.forEach(function(response) {
+      console.log("imported", response.body.title, "HTTP", response.status);
```
Comment (leplatrem) [inline src/js/flux.js:124]: The batch response status might well be always 200. Whereas the status of each response in its body could be 400, 409...

Comment (leplatrem) [inline src/js/flux.js:124]: Sorry for the noise. Just read again your code, and it looks fine.

File: src/test/app_test.js [hunk 3]
```diff
@@ -61,7 +61,31 @@ describe("ArticleList", function() {
...
+  });
+
+  it("should render an import button when the list is empty", function() {
+    var view = TestUtils.renderIntoDocument(
+      <ArticleList articles={[]} />);
+
+    expect($(view, ".list-empty button")).not.eql(null);
+  });
+
+  it("should not render an import button when the list is non-empty", function() {
```
Comment (leplatrem) [inline src/test/app_test.js:74]: As long as importing does not erase existing records, the import button could stay no ?

Comment (n1k0) [inline src/test/app_test.js:74]: Ah, from a UX perspective, I thought it made sense to render it only when the list is empty… importing samples on top of an existing reading list would be rather strange imho.

Comment (leplatrem) [inline src/test/app_test.js:74]: Yes, with samples you're right... Let's keep that for later when importing local files for instance :)

Comment (n1k0) [inline src/test/app_test.js:74]: Sure thing!

**Comment by n1k0** [comment]
Considering this landable, hence landing.

**Pull request merged**